### PR TITLE
Add standalone musl compat libs

### PR DESCRIPTION
**Pull request by alfredfo**
These packages contain common glibc headers that are not available in musl. This is purely for user convenience.

### CODE REVIEW

File: sys-libs/cdefs-standalone/files/sys/cdefs.h [hunk 1]
```diff
@@ -0,0 +1,40 @@
+// taken from https://github.com/void-linux/void-packages/blob/master/srcpkgs/musl-legacy-compat/files/cdefs.h
```
Comment (thesamesam) [inline sys-libs/cdefs-standalone/files/sys/cdefs.h:1]: honestly I'd prefer it if we made a git repo for each of these (same rationale as for tree except for size)

File: sys-libs/tree-standalone/tree-standalone-1.0.0.ebuild [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+
+LICENSE="|| ( BSD-2 BSD )"
+SLOT="0"
+KEYWORDS="~amd64"
+SRC_URI="https://raw.githubusercontent.com/void-linux/void-packages/master/srcpkgs/musl-legacy-compat/files/tree.h"
+
+S="${WORKDIR}"
+
+src_unpack() {
+	cp "${DISTDIR}/${PN%%-*}.h" "${WORKDIR}"
```
Comment (thesamesam) [inline sys-libs/tree-standalone/tree-standalone-1.0.0.ebuild:17]: || die

File: sys-libs/tree-standalone/tree-standalone-1.0.0.ebuild [hunk 3]
```diff
@@ -0,0 +1,22 @@
...
+
+EAPI=8
+
+DESCRIPTION="Compatibility library for musl"
+HOMEPAGE="https://github.com/void-linux/void-packages/tree/master/srcpkgs/musl-legacy-compat"
+
+LICENSE="|| ( BSD-2 BSD )"
+SLOT="0"
+KEYWORDS="~amd64"
+SRC_URI="https://raw.githubusercontent.com/void-linux/void-packages/master/srcpkgs/musl-legacy-compat/files/tree.h"
```
Comment (thesamesam) [inline sys-libs/tree-standalone/tree-standalone-1.0.0.ebuild:12]: SRC_URI should be directly under HOMEPAGE

Comment (thesamesam) [reply sys-libs/tree-standalone/tree-standalone-1.0.0.ebuild:12]: also, didn't we agree not to do this? :)

Comment (alfredfo) [reply sys-libs/tree-standalone/tree-standalone-1.0.0.ebuild:12]: yes, but I just pushed what I had. Will start fixing things now

File: sys-libs/tree-standalone/tree-standalone-1.0.0.ebuild (hunk 2 above)
Comment (alfredfo) [reply sys-libs/tree-standalone/tree-standalone-1.0.0.ebuild:17]: Do you prefer this over just doing doheader from DISTDIR in src_install()?

and then empty src_unpack so it doesn't try to unpack stuff.

File: sys-libs/cdefs-standalone/files/sys/cdefs.h (hunk 1 above)
Comment (alfredfo) [reply sys-libs/cdefs-standalone/files/sys/cdefs.h:1]: sure

**Pull request closed without merging**